### PR TITLE
test: improve transport module unit test coverage

### PR DESCRIPTION
**Pull request by sanity**
## Problem

The transport module had significant gaps in unit test coverage:
- `crypto.rs`: ~50% line coverage - many core functions untested
- `rate_limiter.rs`: ~69% line coverage - no edge case or property-based tests
- No `Arbitrary` implementations for transport types, limiting fuzz testing potential

This was identified using `cargo llvm-cov` to analyze line-by-line coverage.

## This Solution

Add comprehensive unit tests and property-based testing infrastructure:

### crypto.rs (8 new tests)
- `Arbitrary` impls for `TransportKeypair` and `TransportPublicKey` with caching to avoid expensive RSA generation
- Tests for previously uncovered functions: `save()`, `Default`, `new_with_rng()`, `from_private_key()`, `Debug` impl, `From<RsaPublicKey>`
- Keypair save/load round-trip test via PKCS8 PEM encoding

### peer_connection.rs
- `Arbitrary` impl for `StreamId` to enable property-based testing

### rate_limiter.rs (9 new tests)
Property-based tests using the `arbitrary` crate:
- `property_bandwidth_invariant_maintained`: verifies `current_bandwidth` always matches packet sum
- `property_can_send_when_under_limit`: verifies immediate send when under bandwidth limit  
- `property_wait_time_never_exceeds_window`: verifies wait times are bounded by window
- `property_cleanup_removes_old_packets`: verifies expired packets are cleaned up
- `property_partial_cleanup_preserves_recent`: verifies recent packets survive cleanup

Edge case tests:
- Zero packet size, zero bandwidth limit, exact bandwidth limit, very small window

## Testing

All 63 transport tests pass:
```
cargo test --package freenet --lib -- transport::
test result: ok. 63 passed; 0 failed; 5 ignored
```

## Changes

- 439 new lines of test code
- No changes to production code

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/core/src/transport/crypto.rs [hunk 1]
```diff
@@ -156,6 +156,37 @@ impl TransportSecretKey {
     }
 }
 
+#[cfg(test)]
+thread_local! {
+    static CACHED_KEYPAIR: std::cell::RefCell<Option<TransportKeypair>> = const { std::cell::RefCell::new(None) };
```
Comment (iduartgomez) [inline crates/core/src/transport/crypto.rs:161]: we added this back heh

I think expensive calls where also for simulations which we are now not running from fdev (at some point that has to be fixed to catch logical errors in large networks, which should be a more scalable approach to what we are doing right now for testing).

**Comment by sanity** [comment]
The `six-peer-regression` test failure is unrelated to this PR - it's a known flaky integration test that has been failing on main branch as well. This PR only adds test code (439 new lines) and doesn't modify any production code.

Retriggering CI.
